File: eel-core/src/test/kotlin/io/eels/PathIteratorTest.kt
package io.eels

import io.kotlintest.specs.WordSpec
import org.apache.hadoop.fs.Path

class PathIteratorTest : WordSpec() {
  init {
    "PathIterator" should {
      "return all non-null parent paths" {
        val path = Path("/some/path/for/fun")
        PathIterator(path).asSequence().toSet() shouldBe setOf(Path("/some/path/for/fun"), Path("/some/path/for"), Path("/some/path"), Path("/some"), Path("/"))
      }
    }
  }
}

File: eel-core/src/test/scala/io/eels/PathIteratorTest.scala
package io.eels

import org.apache.hadoop.fs.Path
import org.scalatest.{Matchers, WordSpec}

class PathIteratorTest extends WordSpec with Matchers {
  "PathIterator" should {
    "return all non-null parent paths" in {
      val path = new Path("/some/path/for/fun")
      PathIterator(path).toSet shouldBe Set(new Path("/some/path/for/fun"), new Path("/some/path/for"), new Path("/some/path"), new Path("/some"), new Path("/"))
    }
  }
}